### PR TITLE
feat(changes): add initial implementation

### PR DESCRIPTION
**Pull request by etolbakov**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

Implement the `changes` function resets from PromQL(references to [promql's document](https://prometheus.io/docs/prometheus/latest/querying/functions/#changes) and [the go implementation](https://github.com/prometheus/prometheus/blob/90b2f7a540b8a70d8d81372e6692dcbb67ccbaaa/promql/functions.go#L1023-L1040))

## Checklist

- [x]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link:
https://github.com/GreptimeTeam/greptimedb/issues/1226

### CODE REVIEW

File: src/promql/src/functions/changes.rs [hunk 1]
```diff
@@ -12,5 +12,107 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//! Implementation of [`reset`](https://prometheus.io/docs/prometheus/latest/querying/functions/#changes) in PromQL. Refer to the [original
+//! Implementation of [`changes`](https://prometheus.io/docs/prometheus/latest/querying/functions/#changes) in PromQL. Refer to the [original
```
Comment (waynexia) [inline src/promql/src/functions/changes.rs:15]: 🤣 good catch

File: src/promql/src/functions/changes.rs [hunk 2]
```diff
@@ -12,5 +12,107 @@
...
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::functions::test_util::simple_range_udf_runner;
+
+    // build timestamp range and value range arrays for test
+    fn build_test_range_arrays(
```
Comment (waynexia) [inline src/promql/src/functions/changes.rs:57]: We may extract this util into a common function in the follow-up PR

**Review by waynexia — APPROVED** [review]
Good work, thanks ❤️

**Review by v0y4g3r — APPROVED** [review]
Good job

**Pull request merged**